### ISSUE TITLE
cli: Support for large webpack stats files

### ISSUE THREAD

**Issue by vio** [relative-ci/bundle-stats#2118]

**Comment by fcano-uz** [comment]
I'm using the cli to try to generate a stats file, but I get errors like this:

```
Invalid webpack stats structure, please make sure webpack stats configuration is correct.

Expected a value of type `interface<{name,size}> | interface<{type,filteredChildren,size}>[] & notEmptyArray` for `assets`
https://github.com/relative-ci/bundle-stats/tree/master/packages/cli#webpack-configuration
Filepath: ./webpack-stats.json
```

My stats file is +120k lines long. Do you think it can be related to this issue?

**Comment by fcano-uz** [comment]
Update: also I should mention that my `webpack-stats.json` file doesn't contain an `assets` key. (probably because I'm not using any asset). I should also mention I'm on webpack 4. Any of this things might be being the issue, but just in case I'm checking the CLI issues, because with the webpack plugin I didn't had any issue...

**Comment by fcano-uz** [comment]
My stats seem correct, If I log `webpackConfig.stats` (just before exporting it in the config file), I get this:

```js
{
  assets: true,
  entrypoints: true,
  chunks: true,
  modules: true,
  builtAt: true,
  hash: true
}
```

Setting `stats.preset` gives me an "Invalid configuration object" error, I think that might be a webpack 5 option only... 🤔 

I was generating the stats file this way:

```
webpack --json > ./webpack-stats.json
```

I'm seeing there's also a `--profile` option that adds more information about modules in the stats file (is what made my file grow from 70k to 140k lines), but after using that option I still get the same error.

(For reference: this is the [webpack v4 docs for "stats"](https://v4.webpack.js.org/api/stats/))

So even setting the `stats` object as seen in the documentation, and even using the `--profile` option, I get the error when using the CLI.

My stats file has this structure:

![Screenshot 2022-07-08 at 09 32 05](https://user-images.githubusercontent.com/97605542/177940920-db73d126-30c9-4a9e-9a03-a356bc44f8a0.png)

I can open a new issue if you prefer, maybe I can share the full stats file if that can be useful. For now this is a non-blocker, I'm going to use the webpack plugin.

**Comment by vio** [comment]
> Setting stats.preset gives me an "Invalid configuration object" error, I think that might be a webpack 5 option only... 

Yes, you are right about that! On webpack v4, the preset values are options for `stats` (ex: `stats: 'all'`).

bundle-stats supports webpack v4, and I'm still working on a couple of webpack v4 projects where it is used as a standalone CLI or with relative-ci. In this case, I suspect a config conflict with another third-party plugin.

Next, it will be interesting to see the output of the webpack stats with `stats: 'all'` and/or try to use [webpack-stats-plugin](https://github.com/FormidableLabs/webpack-stats-plugin) to output the JSON file.

Let me know if you are still having problems after you try the above. If the assets are still not present, it might be helpful to try to recreate a minimal reproducible case.

**Comment by fcano-uz** [comment]
Thanks for the comment @vio 

Indeed using the [webpack-stats-plugin](https://github.com/FormidableLabs/webpack-stats-plugin) fixes the stats file and makes it work again, so this looks like a plugin overriding the stats configuration.

**Comment by vio** [comment]
closed by #3420 

ready for testing on `v4.4.0-beta.3`
